### PR TITLE
[UI / Charts] Consider default persistence service

### LINKED ISSUE #5993: [UI / Charts] Consider default persistence configuration for drawing charts
Currently the `DefaultChartProvider` uses the first persistence service if one does not define a service for the Chart element. We have a configuration for a default persistence service in the `PersistenceServiceRegistry` which should be considered for querying the data in such case.

https://github.com/eclipse/smarthome/blob/fadc8ef4e5e8ba826bdb14105224086634ea8688/bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/chart/defaultchartprovider/DefaultChartProvider.java#L209-L215

### PR DESCRIPTION
**Pull request by cweitkamp**
- Consider default persistence service
- Added dependency to `PersistenceServiceRegistry` instead of collecting the services on its own

Closes #5993 

// CC @triller-telekom 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by triller-telekom** [comment]
BTW: The travis failure is unrelated and I have documented it here: https://github.com/eclipse/smarthome/issues/5940#issuecomment-409830382

**Comment by kaikreuzer** [comment]
This time, travis failure seems to be related, so please have a look at it:
```
[ERROR] /home/travis/build/eclipse/smarthome/bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/chart/defaultchartprovider/DefaultChartProvider.java:[289] 
[ERROR] 	double convertData(State state) {
[ERROR] 	                   ^^^^^
[ERROR] State cannot be resolved to a type
[ERROR] /home/travis/build/eclipse/smarthome/bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/chart/defaultchartprovider/DefaultChartProvider.java:[327] 
[ERROR] 	State state = null;
[ERROR] 	^^^^^
[ERROR] State cannot be resolved to a type
```

**Comment by kaikreuzer** [comment]
Never heard of IDEs that have an option to remove required imports 😲 ...

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/chart/defaultchartprovider/DefaultChartProvider.java [hunk 1]
```diff
@@ -141,13 +134,25 @@ public String getName() {
...
+                : persistenceServiceRegistry.get(serviceId);
+
+        // Did we find a service?
+        QueryablePersistenceService persistenceService = (service instanceof QueryablePersistenceService)
+                ? (QueryablePersistenceService) service
+                : (QueryablePersistenceService) persistenceServiceRegistry.getAll() //
+                        .stream() //
+                        .filter(it -> it instanceof QueryablePersistenceService) //
+                        .findFirst() //
+                        .orElseThrow(() -> new IllegalArgumentException("No Persistence service found."));
```
Comment (maggu2810) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/chart/defaultchartprovider/DefaultChartProvider.java]: FYI: To filter a stream for specific instances a combination of filter and map could be useful:

```Java
stream().
  filter(QueryablePersistenceService.class::isInstance)
  .map(QueryablePersistenceService.class::cast)
  .findFirst()
```

**Pull request merged**